### PR TITLE
Adding a script to verify api reference docs are upto date

### PR DESCRIPTION
**Pull request by nikhiljindal**
Ref https://github.com/kubernetes/kubernetes/issues/14069

I am not running the verify script anywhere for now, since update-api-reference-docs does not work as expected (https://github.com/kubernetes/kubernetes/issues/15905). 
Will send another PR to start running the script once that issue is fixed.

### CODE REVIEW

File: hack/update-api-reference-docs.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
```
Comment (nikhiljindal) [inline hack/update-api-reference-docs.sh]: Strictly speaking, we do not need this file.
I added it to keep the hack/verify-A.sh, hack/update-A.sh convention.

File: hack/gen-swagger-doc/run-gen-swagger-docs.sh [hunk 2]
```diff
@@ -18,9 +18,14 @@ set -o errexit
...
 set -o pipefail
 
-KUBE_ROOT=$(dirname "${BASH_SOURCE}")/../..
-V1_PATH="$PWD/${KUBE_ROOT}/docs/api-reference/v1/"
-V1BETA1_PATH="$PWD/${KUBE_ROOT}/docs/api-reference/extensions/v1beta1"
+if [ "$#" -lt 1 ]; then
+    echo "Usage: run-gen-swagger-docs.sh <absolute output path>"
+    exit
+fi
+OUTPUT=${1:-}
```
Comment (caesarxuchao) [inline hack/gen-swagger-doc/run-gen-swagger-docs.sh]: Why not setting the default value for OUTPUT to "$PWD/${KUBE_ROOT}/docs/api-reference/"? That will be convenient.

Comment (caesarxuchao) [inline hack/gen-swagger-doc/run-gen-swagger-docs.sh]: Never mind. I see you add another script to call this script.

File: hack/update-api-reference-docs.sh (hunk 1 above)
Comment (caesarxuchao) [inline hack/update-api-reference-docs.sh]: How about move the hack/gen-swagger-doc/run-gen-swagger-docs.sh to hack/ and rename it? I feel there are too many layers.

**Comment by caesarxuchao** [comment]
Thank you @nikhiljindal! LGTM except one nit.

File: hack/update-api-reference-docs.sh (hunk 1 above)
Comment (nikhiljindal) [inline hack/update-api-reference-docs.sh]: Done

**Comment by nikhiljindal** [comment]
Have moved gen-swagger-docs/run-gen-swagger-docs.sh to hack/update-api-reference-docs.sh
Also added a commit by running that script so that verify-api-reference-docs.sh passes.

@caesarxuchao PTAL.

**Comment by caesarxuchao** [comment]
Thank you @nikhiljindal, LGTM. Do we need to block this one before we install docker in shippable?

**Comment by nikhiljindal** [comment]
No. This does not change what is run on shippable, so we should be fine.

**Pull request merged**